pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/OfflineClusterIntegrationTest.java

```diff
@@ -598,6 +598,59 @@ public void testSelectionUDF()
     }
   }
 
+  @Test
+  public void testFilterUDF()
```

Comment:
Can we have more test cases to cover, e.g. nested expressions in filter?

Reply:
Added one more test case.